.github/workflows/smile-server.yaml

```diff
@@ -70,18 +70,30 @@ jobs:
             VERSION="$BRANCH"
             tags="$BRANCH"
           fi
+          
+          echo "GITHUB_BRANCH=${BRANCH}" >> $GITHUB_ENV
 
           # should we push to dockerhub, and is there a README
           DOCKERHUB_PUSH="false"
           DOCKERHUB_README="false"
           if [ "${{ github.repository }}" == "${{ env.MAIN_REPO }}" ]; then
-            if [ "${{ secrets.DOCKERHUB_USERNAME }}" != "" -a "${{ secrets.DOCKERHUB_PASSWORD }}" != "" ]; then
-              DOCKERHUB_PUSH="true"
-              if [ -e "README.md" ]; then
-                DOCKERHUB_README="true"
+            if [ ${BRANCH} == "main" ]; then
+              if [ "${{ secrets.DOCKERHUB_USERNAME }}" != "" -a "${{ secrets.DOCKERHUB_PASSWORD }}" != "" ]; then
+                DOCKERHUB_PUSH="true"
+                if [ -e "README.md" ]; then
+                  DOCKERHUB_README="true"
+                fi
               fi
             fi
           fi
+          
+          # should we push to Github
+          GITHUB_PUSH="false"
+          if [ "${{ github.repository }}" == "${{ env.MAIN_REPO }}" ]; then
+            if [ ${BRANCH} == "develop" ]; then
```

Comment:
I think it's everything else could go to github ghrc. Maybe remove the "develop" match?

Reply:
oh okay, so we want to put everything in github ghrc 
The other analytics is like
merge to main: docker hub
merge to develop: ghrc
others: done by manual

but you want 
merge to main: docker hub
any push: grhc

right? Just confirming

Reply:
Right. Cuz analytics have manual triggers that we can use to push to github.

Reply:
Okay, it is updated. So all the PR's will go to github repo